src/forcing/ForcingsEngineLumpedDataProvider.cpp

```diff
@@ -0,0 +1,259 @@
+        }
+    }
+}
+
+double Provider::at(
+    const std::string& divide_id,
+    const std::string& variable,
+    bool               previous
+)
+{
+    const auto d_idx = divide_index(divide_id);
+    const auto v_idx = variable_index(variable);
+
+    if (d_idx == bad_index || v_idx == bad_index) {
+        const auto shape = var_cache_.shape();
+        const auto shape_str = std::to_string(shape[0]) + ", " +
+                               std::to_string(shape[1]) + ", " +
+                               std::to_string(shape[2]);
+
+        const auto index_str = (d_idx == bad_index ? "?" : std::to_string(d_idx)) + ", " +
+                               (v_idx == bad_index ? "?" : std::to_string(v_idx));
+
+        throw std::out_of_range{
+            "Failed to get ForcingsEngineLumpedDataProvider value at index {" + index_str + "}" +
+            "\n  Shape    : {" + shape_str + "}" +
+            "\n  Divide   : `" + divide_id + "`" +
+            "\n  Variable : `" + variable  + "`"
+        };
+    }
+
+    return at(d_idx, v_idx, previous);
+}
+
+double Provider::at(
+    std::size_t divide_idx,
+    std::size_t variable_idx,
+    bool      previous
+)
+{
+    if (divide_idx == bad_index || variable_idx == bad_index) {
+        const auto shape = var_cache_.shape();
+        const auto shape_str = std::to_string(shape[0]) + ", " +
+                               std::to_string(shape[1]) + ", " +
+                               std::to_string(shape[2]);
+
+        const auto index_str = (divide_idx == bad_index ? "?" : std::to_string(divide_idx)) + ", " +
+                               (variable_idx == bad_index ? "?" : std::to_string(variable_idx));
+
+        throw std::out_of_range{
+            "Failed to get ForcingsEngineLumpedDataProvider value at index {" + index_str + "}" +
+            "\n  Shape          : {" + shape_str + "}"
+        };
+    }
+    
+    return var_cache_.at({{previous ? 1U : 0U, divide_idx, variable_idx}});
+}
+
+double Provider::get_value(
+    const CatchmentAggrDataSelector& selector,
+    ReSampleMethod m
+)
+{
+    const auto start = clock_type::from_time_t(selector.get_init_time());
+    const auto end = std::chrono::seconds{selector.get_duration_secs()} + start;
+    const auto step = std::chrono::seconds{this->record_duration()};
+    const std::string id = selector.get_id();
+    const std::string var = selector.get_variable_name();
+
+    if (m == ReSampleMethod::SUM || m == ReSampleMethod::MEAN) {
+        double acc = 0.0;
+        auto current_time = start;
+        while (current_time < end) {
+            if (!this->next()) {
+                break;
+            }
+
+            acc += this->at(id, var);
+            current_time += step;
+        }
+
+        if (m == ReSampleMethod::MEAN) {
+            const auto time_step_seconds = step.count();
+            const auto time_duration = std::chrono::duration_cast<std::chrono::seconds>(current_time.time_since_epoch()).count();
```

Comment:
```suggestion
            const auto time_duration = std::chrono::duration_cast<std::chrono::seconds>(current_time - start).count();
```
I think I'm misunderstanding, but should this be relative to `start` and not the common epoch?